src/robocoup/commands/utilization.js

```diff
@@ -1,25 +1,183 @@
+    const nonBillableYtd = R.filter(isNonBillable)(myYtd);
+    const nonBillableYtdUsage = R.filter(hasDays)(nonBillableYtd);
+    const myYtdMetrics = [
+      `> *My Utilization Status: ${moment().year()} To Date*`,
+      `> ${ytdBar(getTotal(billableYtd), 'Billable')}`,
+      billableYtdUsage.map(metric => `>╰ ${ytdBar(metric.total, metric.name)}`),
+      `> ${ytdBar(getTotal(nonBillableYtd), 'Non-Billable')}`,
+      nonBillableYtdUsage.map(metric => `>╰ ${ytdBar(metric.total, metric.name)}`),
+    ];
+
+    // calculate team member utilization for the last six months
+    const billablePast = R.filter(isBillable)(past);
+    const billablePastUsage = R.filter(hasDays)(billablePast);
+    const nonBillablePast = R.filter(isNonBillable)(past);
+    const nonBillablePastUsage = R.filter(hasDays)(nonBillablePast);
+    const myPastMetrics = [
+      `> *My Utilization Status: Last Six Months*`,
+      `> ${pastBar(getTotal(billablePast), 'Billable')}`,
+      billablePastUsage.map(metric => `>╰ ${pastBar(metric.total, metric.name)}`),
+      `> ${pastBar(getTotal(nonBillablePast), 'Non-Billable')}`,
+      nonBillablePastUsage.map(metric => `>╰ ${pastBar(metric.total, metric.name)}`),
+    ];
+
+    // calculate team member utilization for the next six months
+    const billableFuture = R.filter(isBillable)(future);
+    const billableFutureUsage = R.filter(hasDays)(billableFuture);
+    const nonBillableFuture = R.filter(isNonBillable)(future);
+    const nonBillableFutureUsage = R.filter(hasDays)(nonBillableFuture);
+    const myFutureMetrics = [
+      `> *My Utilization Status: Next Six Months* _(as currently scheduled)_`,
+      `> ${futureBar(getTotal(billableFuture), 'Billable')}`,
+      billableFutureUsage.map(metric => `>╰ ${futureBar(metric.total, metric.name)}`),
+      `> ${futureBar(getTotal(nonBillableFuture), 'Non-Billable')}`,
+      nonBillableFutureUsage.map(metric => `>╰ ${futureBar(metric.total, metric.name)}`),
+      `> ${futureBar(countDaysFuture - getTotal(future), 'Unscheduled')}`,
+    ];
+
+    // calculate which utilization/leave types are possible
+    const types = R.uniq(R.map(R.prop('name'))(myYearWindow));
+    // calculate which utilizations/leave types a team member has used
+    const nonEmptyTypes = R.uniq(R.concat(
+      R.map(R.prop('name'))(R.filter(hasDays)(myYearWindow)),
+      R.map(R.prop('name'))(R.filter(hasDays)(myYtd))
+    ));
+    // calculate which utilization/leave types haven't been used
+    const unusedTypes = R.difference(types, nonEmptyTypes).sort();
+    const myUtilizationTypesNotUsed = [
+      `> *You have no utilizations for the following types:*`,
+      `> ${unusedTypes.join(', ')}`,
+    ];
+
+    const resources = [
+      `> *Resources*`,
+      `> <https://github.com/bocoup/bocoup-meta/wiki/Policies#time-off|Time Off Policy>`,
+      `> <https://away.bocoup.com|Make a Time Off Request>`,
+    ];
+
     return [
```

Comment:
You sure there isn't some Ramda function to insert spaces between array elements? heh